opal/corelib/io.rb

```diff
@@ -3,6 +3,55 @@ class IO
   SEEK_CUR = 1
   SEEK_END = 2
 
+  attr_reader :eof
+  attr_reader :lineno
+
+  def initialize
+    @eof = false
+    @lineno = 0
+  end
+
+  def read
+    if @eof
+      ''
+    else
+      res = File.read(@path)
+      @eof = true
+      @lineno = res.size
+      res
+    end
+  end
+
+  def each_line(separator = $/, &block)
+    if @eof
+      return block_given? ? self : [].to_enum
+    end
+
+    if block_given?
+      lines = File.read(@path)
+      %x(
```

Comment:
Can you change this to curly? (just for consistency with the rest of the codebase)